python/ray/_private/accelerator.py

```diff
@@ -107,7 +107,7 @@ def _detect_and_configure_custom_accelerator(
         # Don't use more resources than allowed by the user's pre-set values.
         if accelerator_count is not None and visible_ids is not None:
             accelerator_count = min(accelerator_count, len(visible_ids))
-    if accelerator_count is not None:
+    if accelerator_count is not None and accelerator_count > 0:
```

Comment:
@rkooo567 can you help decide if this change is okay for AWS NeuronCore (or do you know who would know?)

It just means that 

```
        resources.update(
            {
                accelerator_key: accelerator_count,
                get_accelerator_type(): accelerator_count,
            }
        )
```

will no longer be called if neuron cores are detected but the number is 0 (not sure if this situation is possible.)

Reply:
It seems the situation is possible. The neuron count can be `0` if `sys.platform.startswith("linux") and os.path.isdir("/opt/aws/neuron/bin/")`, otherwise it's `None`.  

Still, I think it's unlikely that anyone is depending on the behavior of `resources` being updated with `{<neuron key>: 0}`, especially because it's a new experimental API.

Reply:
yeah resource :0 will anyway be ignored from core I think. This must be okay.